pkgs/os-specific/darwin/apple-sdk-12.3/libcharset.nix

```diff
@@ -0,0 +1,20 @@
+{
+  stdenvNoCC,
+  buildPackages,
+  darwin-stubs,
+}:
+
+stdenvNoCC.mkDerivation {
```

Comment:
runCommand ? maybe not due to the hook

Reply:
I cleaned up the derivations to use `buildCommand` and copy the files to avoid requiring the full set of stubs for a tiny header. For darwin-stubs, I ended up setting `preferLocalBuild = true` and `allowSubstitutes = false`, which is all that `runCommandLocal` does.